src/client/src/services/limitChecker/limitChecker.openfin.ts

```diff
@@ -22,37 +22,48 @@ export const checkLimit$: CheckLimitStreamGenerator = (message: {
   tradedCurrencyPair: string
   notional: number
   rate: number
-}) => {
-  const payload = {
-    notional: message.notional.toString(),
-    rate: message.rate.toString(),
-    tradedCurrencyPair: message.tradedCurrencyPair,
-    id: (limitCheckId++).toString(),
-  }
-
-  const result$ = new Subject<string>()
-
-  const listener = window.fdc3.addContextListener(
-    `result-${payload.id}`,
-    (context) => {
-      if (context.id) {
-        result$.next(context.id.result)
+}) =>
+  isLimitCheckerRunning$.pipe(
```

Comment:
This was where the bug was, we need some condition in the stream to check whether the limit checker is open, it existed in the previous version in a similar form to this I just never put it back in.